docs/architecture/session.md

```diff
@@ -0,0 +1,66 @@
+# Session
+
+__Session__ is the actor that encapsulates the logic of the sessions 
+associated to each TCP connections.
```

Comment:
- This actor actually encapsulates the entire business logic of the Witnet network protocol.
- This actor shouldn't care about the specific transport used by the `ConnectionsManager`, ans so shouldn't its documentation.